02/PrintRecords.cpp

```diff
@@ -7,95 +7,141 @@ using namespace std;
 
 namespace samples
 {
-	void InputRecords(Record* info)
+	Record InputRecord(Record record)
 	{
 		cout << "First name: ";
-		cin >> info->FirstName;
+		cin >> record.FirstName;
 		
 		cout << "Last name: ";
-		cin >> info->LastName;
+		cin >> record.LastName;
 		
 		cout << "Student ID: ";
-		cin >> info->StudentID;
+		cin >> record.StudentID;
 
 		cout << "Score:  ";
-		cin >> info->Score;
+		cin >> record.Score;
+
+		return record;
 	}
 
 	void ManageRecordsExample()
 	{
-		Record studentInfos[5];
-		ofstream fileStream("studentRecords.dat", ios_base::out);
+		Record record;
+
+		ofstream outputStream("studentRecords.dat");
+		ifstream inputStream("studentRecords.dat");
 
-		int existingStudents = 1;
+		int recordCount = 0;
 
-		studentInfos[0].FirstName = "Pope";
-		studentInfos[0].LastName = "Kim";
-		studentInfos[0].StudentID = "A12345678";
-		studentInfos[0].Score = 80;
+		record.FirstName = "Pope";
+		record.LastName = "Kim";
+		record.StudentID = "A12345678";
+		record.Score = "80";
 
-		PrintRecords(&studentInfos[0], &fileStream, existingStudents);
+		WriteFileRecord(&outputStream, record, true, &recordCount);
 
 		while (true)
 		{
 			char input;
 
-			cout << "a: adds another student info. " << "b : Modifies existing info. " << endl;
+			cout << "a: add a student record. " << "b: Modify a student record. " << "x: exit" << endl;
 			cin >> input;
```

Comment:
my question on this code is... what happens to the new character?

1. user enters "a\n" 
2. input = 'a'  
3. still \n is in cin? or it is discarded?